eventsource-client/src/client.rs

```diff
@@ -212,12 +229,13 @@ impl State {
             State::Connecting { retry: true, .. } => "connecting(retry)",
             State::Connected(_) => "connected",
             State::WaitingToReconnect(_) => "waiting-to-reconnect",
+            State::StreamClosed => "closed",
         }
     }
 }
 
 impl Debug for State {
-    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
```

Comment:
Why did you have to add that lifetime?

Reply:
It's a lint that got enabled when I added `#![warn(rust_2018_idioms)]`. It's debatable whether or not we need these, but since we are using the 2018 edition it seemed prudent. 

```
warning: hidden lifetime parameters in types are deprecated
   --> eventsource-client/src/client.rs:238:27
    |
238 |     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
    |                           ^^^^^^^^^- help: indicate the anonymous lifetime: `<'_>`
```

Reply:
Good to know! Thanks!